client/data/marketplace/use-site-search-es-query.ts

```diff
@@ -0,0 +1,217 @@
+import phpUnserialize from 'phpunserialize';
+import { UseQueryResult, UseQueryOptions, InfiniteData, useInfiniteQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import { extractSearchInformation, normalizePluginsList } from 'calypso/lib/plugins/utils';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import {
+	DEFAULT_FIRST_PAGE,
+	DEFAULT_PAGE_SIZE,
+	DEFAULT_CATEGORY,
+	BASE_STALE_TIME,
+	SITE_SEARCH_CACHE_KEY,
+	SITE_SEARCH_ENDPOINT,
+} from './constants';
+import { ESHits, ESResponse, Plugin, PluginQueryOptions } from './types';
+import { getPluginsListKey } from './utils';
+
+/**
+ *
+ * @param pluginSlug
+ * @param iconsArray
+ * @returns An object containing a resolution -> icon url map
+ */
+const createIconsObject = ( pluginSlug: string, iconsArray: string ) => {
+	type Icon = {
+		resolution: string;
+		filename: string;
+		location: string;
+		revision: string;
+	};
+	const icons: Record< string, Icon > = phpUnserialize( iconsArray );
+	return Object.values( icons ).reduce(
+		( prev: Record< string, string >, { resolution, filename, location, revision } ) => {
+			prev[
+				resolution
+			] = `https://ps.w.org/${ pluginSlug }/${ location }/${ filename }?rev=${ revision }`;
+			return prev;
+		},
+		{}
+	);
+};
+
+const createAuthorUrl = ( header_author: string, header_author_uri: string ) =>
+	`<a href="${ header_author_uri }">${ header_author }</a>`;
+
+const mapIndexResultsToPluginData = ( results: ESHits ): Plugin[] => {
+	return results.map( ( { fields: hit } ) => {
+		return {
+			name: hit.title_en, // TODO: add localization
+			slug: hit.slug,
+			version: hit.stable_tag,
+			author: createAuthorUrl(
+				hit[ 'meta.header_author.value' ],
+				hit[ 'meta.header_author_uri.value' ]
+			),
+			author_profile: hit[ 'meta.header_author_uri.value' ], // TODO: get author profile URL
+			tested: hit.tested,
+			rating: hit.rating,
+			num_ratings: hit.num_ratings,
+			support_threads: hit.support_threads,
+			support_threads_resolved: hit.support_threads_resolved,
+			active_installs: hit.active_installs,
+			last_updated: hit[ 'meta.last_updated.value' ],
+			short_description: hit.excerpt_en, // TODO: add localization
+			icons: createIconsObject( hit.slug, hit[ 'meta.assets_icons.value' ] ),
+		};
+	} );
+};
+
+/**
+ *
+ * @param url The url to post to
+ * @param body The body of the request
+ * @returns {Promise<Response>} A promise that resolves to the response body
+ */
+async function postRequest( url: string, body: object ) {
+	const response = await fetch( `${ url }`, {
+		method: 'POST',
+		body: JSON.stringify( body ),
+		headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
+	} );
+
+	if ( response.ok ) {
+		return await response.json();
+	}
+	throw new Error( await response.text() );
+}
+
+export async function fetchWPOrgPluginsFromSiteSearch( options: PluginQueryOptions ) {
```

Comment:
Nitipinking

The ES index will have only WPOrg plugins for now?

Maybe it is a good idea to use a more generalized method name as it should come from the same place(unified index) in the future, and terms as WPcom and WPorg plugins won't make sense anymore.

Reply:
AFAIK yes, until D79130-code is landed we won't be able to query other sources.

I agreee that the name should be more generic, I've updated it. 👍